bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerV2Handler.java

```diff
@@ -369,33 +362,35 @@ private void sendData(String key, String value) {
 
             var statusCode = contentResponse.getStatus();
             if (!(statusCode == 200 || statusCode == 202 || statusCode == 204)) {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                        "@text/unsuccessful.communication-error");
+                updateStatus(OFFLINE, COMMUNICATION_ERROR, "@text/unsuccessful.communication-error");
                 logger.debug("Could not send data, Response {}, StatusCode: {}", response, statusCode);
             }
         } catch (InterruptedException ie) {
             Thread.currentThread().interrupt();
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, ie.toString());
+            updateStatus(OFFLINE, COMMUNICATION_ERROR, ie.toString());
             logger.debug("Could not send data: {}, {}", urlStr, ie.toString());
         } catch (TimeoutException | ExecutionException | JsonSyntaxException e) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.toString());
+            updateStatus(OFFLINE, COMMUNICATION_ERROR, e.toString());
             logger.debug("Could not send data: {}, {}", urlStr, e.toString());
         }
     }
 
     /**
-     * Request new data from Go-eCharger
+     * Retrieves data from the Go-E Charger API based on the configured API version.
+     * Sends an HTTP GET request to the charger and parses the response into the appropriate DTO.
      *
-     * @return the Go-eCharger object mapping the JSON response or null in case of
-     *         error
-     * @throws ExecutionException
-     * @throws TimeoutException
-     * @throws InterruptedException
+     * @return A {@link GoEStatusResponseBaseDTO} object containing the parsed response data.
+     *         Returns {@link GoEStatusResponseDTO} for API version 1 and {@link GoEStatusResponseV2DTO} for API version
+     *         2.
+     * @throws InterruptedException If the thread is interrupted while waiting for the response.
+     * @throws TimeoutException If the request times out.
+     * @throws ExecutionException If an exception occurs during the execution of the request.
+     * @throws JsonSyntaxException If the response JSON cannot be parsed into the expected DTO.
+     * @throws IllegalArgumentException If the response JSON is invalid or does not match the expected format.
      */
-    @Nullable
     @Override
-    protected GoEStatusResponseBaseDTO getGoEData()
-            throws InterruptedException, TimeoutException, ExecutionException, JsonSyntaxException {
+    protected @Nullable GoEStatusResponseBaseDTO getGoEData() throws InterruptedException, TimeoutException,
+            ExecutionException, JsonSyntaxException, IllegalArgumentException {
```

Comment:
I don't see what could generate an IllegalArgumentException in this method.

Reply:
The super method that is overriden has this annotated. I kept the same signature.

Reply:
Ah ok